hack/test.sh

```diff
@@ -17,4 +17,6 @@ set -e
 
 echo "> Test"
```

Comment:
Can you maybe add a bit of developer documentation about how this stuff with envtest works?
This could also include a small guide on how to execute the tests directly via ginkgo (set `KUBEBUILDER_ASSETS` env var).